tidy3d/components/structure.py

```diff
@@ -77,6 +77,21 @@ class Structure(AbstractStructure):
         discriminator=TYPE_TAG_STR,
     )
 
+    def eps_diagonal(self, frequency: float) -> Tuple[complex, complex, complex]:
+        """Main diagonal of the complex-valued permittivity tensor as a function of frequency.
+
+        Parameters
+        ----------
+        frequency : float
+            Frequency to evaluate permittivity at (Hz).
+
+        Returns
+        -------
+        complex
+            The diagonal elements of the relative permittivity tensor evaluated at ``frequency``.
+        """
+        return self.medium.eps_diagonal(frequency=frequency)
```

Comment:
For custom medium, I think one way is to define a particular method in custom medium, so it'll be 
```
if isinstance(sim_struct.medium, CustomMedium):
    return self.medium.somemethod(frequency, coords)
return self.medium.eps_diagonal(frequency=frequency)
```

Reply:
yes I think this will be the approach.